functions/src/index.ts

```diff
@@ -46,163 +45,194 @@ const genericErrorHandler = (err: any, req: express.Request, res: express.Respon
-      if (!token) {
-        res.error(400, "Missing token in headers, query or cookie");
-      }
-      else if (token === "test") {
-        req.claims = {
-          user_id: "http://example.com/users/test",
-          platform_user_id: "test",
-          platform_id: "http://example.com",
-          context_id: "testContextId"
-        };
-        next();
-      }
-      else {
-        const public_key = config.admin.public_key.split("\\n").join("\n");
-        verify(token, public_key, { algorithms: ["RS256"] }, (error: any, decoded: any) => {
-          if (error) {
-            res.error(403, error);
-          }
-          else {
-            const claims = decoded.claims as JWTClaims;
-            if (!claims.user_id) {
-              res.error(403, "Missing user_id in JWT claims!");
-            }
-            if (!claims.platform_user_id) {
-              res.error(403, "Missing platform_user_id in JWT claims!");
-            }
-            if (!claims.platform_id) {
-              res.error(403, "Missing platform_id in JWT claims!");
-            }
-            // context_id is optional so don't test for it
-            // TODO: context_id is still "class_hash" rigse jwt_controller.rb - should it be class_hash here?
-            req.claims = claims;
-            next();
-          }
-        })
-      }
-    })
-    .catch(error => res.error(403, error))
+const authUsingJWTOrReadWriteToken = (req: express.Request): JWTClaims | string => {
+  const token = getToken(req);
+  if (!token) {
+    throw new Error("Missing token in headers, query or cookie");
+  } else if (token.startsWith(ReadWriteTokenPrefix)) {
+    return token;
+  } else {
+    return getClaimsFromJWT(token);
+  }
 };
 
 app.use(customHandlers);
 app.use(genericErrorHandler);
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cors());
-app.use(checkAuth);
-
-const checkEnv = (req: express.Request, res: express.Response, next: express.NextFunction) => {
-  const { env } = req.query;
-  if (!env) {
-    res.error(403, "Missing env query parameter!");
-  }
-  else {
-    req.env = env as string;
-    next();
-  }
-};
 app.use(checkEnv);
 
 app.get('/api/v1/resources', (req, res) => {
-  return BaseResourceObject.FindAll(db, req.env, req.claims, req.query)
-    .then(resources => resources.map((resource) => resource.apiResult()))
-    .then(apiResults => res.success(apiResults))
-    .catch(error => res.error(400, error))
-})
+  try {
+    const claims = authUsingJWT(req);
+    BaseResourceObject.FindAll(db, req.env, claims, req.query)
+      .then(resources => resources.map((resource) => resource.apiResult()))
+      .then(apiResults => res.success(apiResults))
+      .catch(error => res.error(400, error))
+  } catch (error) {
+    res.error(403, error)
+  }
+});
 
 app.get('/api/v1/resources/:id', (req, res) => {
-  return BaseResourceObject.Find(db, req.env, req.params.id)
+  BaseResourceObject.Find(db, req.env, req.params.id)
```

Comment:
This doesn't seem to be checking any auth. Doesn't this mean that someone who has the id of the resource would get the full object stored in firestore which would probably include the readWriteToken?

Reply:
Ops, good catch. This was always open. So, I think there are two options:
1. We can lock it. But then the question is if we don't break some apps (glossary, vortex).
2. We can remove accessRules if user is not authorized.

Reply:
So, I tried 2 in one of the new commits. You can take a look if it looks good to you. AccessRules will be stripped depending on auth claims. In the future, we can apply the same approach to other fields if necessary.